pkg/tracker/tracker_test.go

```diff
@@ -0,0 +1,142 @@
+package tracker
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+type request struct {
+	tracker *tracker
+	event   *Event
+}
+
+func TestTracker(t *testing.T) {
+	t.Run("NewTracker", func(test *testing.T) {
+		for _, test := range []struct {
+			name     string
+			gaAPIURl string
+			expected error
+		}{
+			{
+				name:     "GA Tracking ID should not be empty",
+				gaAPIURl: "",
+				expected: errors.New("tracker.NewTracker: GA_TRACKING_ID env not set"),
+			},
+			{
+				name:     "Tracker should initialize properly",
+				gaAPIURl: "/",
+				expected: nil,
+			},
+		} {
+			t.Run(test.name, func(testing *testing.T) {
+				if test.expected == nil {
+					os.Setenv(GA_TRACKING_ID, "UA-000000-2")
+					resp := NewTracker()
+					assert.Nil(testing, resp, test.expected)
+				} else {
+					resp := NewTracker()
+					assert.NotNil(testing, resp)
+					assert.Equal(testing, resp.Error(), test.expected.Error())
+				}
+			})
+		}
+	})
+
+	t.Run("SendEvent", func(test *testing.T) {
+		id, err := uuid.NewV4()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		tracker := &tracker{
+			gaPropertyID: "UA-000000-2",
+			gaAPIURL:     "/",
+			cid:          id.String(),
+		}
+		for _, test := range []struct {
+			name     string
+			request  *request
```

Comment:
I see we are using same tracker all request structure instances. 
Do we really need request type or we should be fine event only for tests ?